services/src/main/java/org/exoplatform/officeonline/AbstractOfficeOnlineService.java

```diff
@@ -162,6 +171,54 @@ protected boolean canEditDocument(Node node) throws RepositoryException {
     return res;
   }
 
+  /**
+   * Gets the content.
+   *
+   * @param fileId the fileId
+   * @param config the config
+   * @return the content
+   * @throws OfficeOnlineException the office online exception
+   */
+  public DocumentContent getContent(String fileId, EditorConfig config) throws OfficeOnlineException {
+    if (!fileId.equals(config.getFileId())) {
+      throw new BadParameterException("FileId doesn't match fileId specified in token");
+    }
+    try {
+      Node node = nodeByUUID(config.getFileId(), config.getWorkspace());
+      Node content = nodeContent(node);
+
+      final String mimeType = content.getProperty("jcr:mimeType").getString();
+      // data stream will be closed when EoF will be reached
+      final InputStream data = new AutoCloseInputStream(content.getProperty("jcr:data").getStream());
+      return new DocumentContent() {
+        @Override
+        public String getType() {
+          return mimeType;
+        }
+
+        @Override
+        public InputStream getData() {
+          return data;
+        }
+
+        @Override
+        public String getVersion() {
+          try {
+            return node.isNodeType("mix:versionable") ? node.getBaseVersion().getName() : JCR_FIRST_VERSION;
```

Comment:
It's not correct to return version '1' as it may mislead in different cases. It will be good to return `null`, or '0', and build client side logic accordingly. When it will be consumed by Office side, then need understand how exactly it will be interpreted/acted and seen by users. If it's mandatory, then version '0' is a good candidate.